src/compiler/scala/tools/nsc/transform/UnCurry.scala

```diff
@@ -215,15 +216,32 @@ abstract class UnCurry extends InfoTransform
      *    new \$anon()
      *
      */
-    def transformFunction(fun: Function): Tree =
+    def transformFunction(fun: Function): Tree = {
+      // in the forwarding expansion def f(ps) = lifted_f(ps) ensure ps never applied
+      def noApplyInExpansion(expansion: Tree): Unit =
+        expansion match {
+          case Block(ClassDef(_, _, _, Template(_, _, body)) :: Nil, _) =>
+            body.last match {
+              case dd: DefDef =>
+                treeInfo.dissectApplied(dd.rhs).argss match {
+                  case args :: _ => args.foreach(noApply.addOne)
```

Comment:
could we put the `noApply` tag directly when the trees are created in `gen.expandFunction`, by making `noApply` a tree attachment?

Reply:
I think that was the previous version? but I'll look again with so-called fresh eyes.